### ISSUE TITLE
Changing the prompt string dynamically

### ISSUE THREAD

**Issue by SirWindfield** [kkawakam/rustyline#417]
Hey!

I have a rather "special" use case. I want the user to enter a string of a specific length `n`. For convenience I wanted to show the remaining characters that they have while typing. Something like this, with `>>` being the prompt:
```
(56) >> User already typed stuff
```

Is this somehow possible?

**Comment by gwenn** [comment]
No, you cannot but you may use `highlight_pompt` as a workaround with some limitations:
https://github.com/kkawakam/rustyline/pull/369#issuecomment-616720954

**Comment by gwenn** [comment]
For reference, `replxx` uses a callback:
https://github.com/AmokHuginnsson/replxx/blob/master/examples/cxx-api.cxx#L228-L232
